cmd/ci-secret-generator/main.go

```diff
@@ -0,0 +1,232 @@
+	if err != nil {
+		return nil, fmt.Errorf("bw cmd failed: %v", string(out))
+	}
+	return out, nil
+}
+
+func updateSecrets(bwItems []bitWardenItem, bwClient bitwarden.Client, dryRun bool) error {
+	var tmpFile *os.File
+	for _, bwItem := range bwItems {
+		if dryRun {
+			var err error
+			tmpFile, err = ioutil.TempFile("", "ci-secret-generator")
+			if err != nil {
+				logrus.WithError(err).Fatal("failed to create tempfile")
+			}
+			defer tmpFile.Close()
+			logrus.Infof("Dry-Run enabled, writing secrets to %s", tmpFile.Name())
+			if _, err := fmt.Fprintln(tmpFile, "---"); err != nil {
+				return err
+			}
+		}
+		if dryRun {
+			if _, err := fmt.Fprintln(tmpFile, "Item: "+bwItem.ItemName); err != nil {
+				return err
+			}
+		}
+
+		if bwItem.Field.Name != "" {
+			out, err := executeCommand(bwItem.Field.Cmd)
+			if err != nil {
+				return fmt.Errorf("failed to execute command: %w", err)
+			}
+			if dryRun {
+				if _, err := fmt.Fprintln(tmpFile, "\tField: "+string(out)); err != nil {
+					return err
+				}
+			} else {
+				if err = bwClient.SetFieldOnItem(bwItem.ItemName, bwItem.Field.Name, out); err != nil {
+					return fmt.Errorf("failed to set field item: %s, field: %s - %w", bwItem.ItemName, bwItem.Field.Name, err)
+				}
+			}
+		}
+		if bwItem.Attachment.Name != "" {
+			out, err := executeCommand(bwItem.Attachment.Cmd)
+			if err != nil {
+				return fmt.Errorf("failed to execute command: %w", err)
+			}
+			if dryRun {
+				if _, err := fmt.Fprintln(tmpFile, "\tAttachment: "+string(out)); err != nil {
+					return err
+				}
+			} else {
+				if err = bwClient.SetAttachmentOnItem(bwItem.ItemName, bwItem.Attachment.Name, out); err != nil {
+					return fmt.Errorf("failed to set attachment, item: %s, attachment: %s - %w", bwItem.ItemName, bwItem.Attachment.Name, err)
+				}
+			}
+		}
+		if bwItem.Attribute.Name != "" {
+			out, err := executeCommand(bwItem.Attribute.Cmd)
+			if err != nil {
+				return fmt.Errorf("failed to create password: %w", err)
+			}
+			if dryRun {
+				if _, err := fmt.Fprintln(tmpFile, "\tPassword: "+string(out)); err != nil {
+					return err
+				}
+			} else {
+				if err = bwClient.SetPassword(bwItem.ItemName, out); err != nil {
+					return fmt.Errorf("failed to set password, item: %s, attachment: %s - %w", bwItem.ItemName, bwItem.Attachment.Name, err)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func main() {
+	// CLI tool which does the secret generation and uploading to bitwarden
+	o := parseOptions()
+	secrets := sets.NewString()
+	logrus.SetFormatter(logrusutil.NewCensoringFormatter(logrus.StandardLogger().Formatter, func() sets.String {
+		return secrets
+	}))
+	if err := o.validateOptions(); err != nil {
+		logrus.WithError(err).Fatal("invalid arguments.")
+	}
+	if err := o.completeOptions(&secrets); err != nil {
+		logrus.WithError(err).Fatal("failed to complete options.")
+	}
+	bwClient, err := bitwarden.NewClient(o.bwUser, o.bwPassword, func(s string) {
+		secrets.Insert(s)
+	})
```

Comment:
The insert function's signature is slightly different from the func(string), since it returns a set and takes in variadic as input. Making this change would force me to change the function signature where its being defined even though the return type is unnecessary. Does it still make sense to proceed with this change?

Reply:
My bad, I didn't know it was variadic.  Never mind, then.